From 4badd709db2234f5e723d3941f2fa03f8b515780 Mon Sep 17 00:00:00 2001
Message-ID: <4badd709db2234f5e723d3941f2fa03f8b515780.1705743628.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 16 Jan 2024 23:53:40 +0000
Subject: [PATCH 22/38] gnu: go-lukechampine-com-blake3: Move to (gnu packages
 golang-crypto).

* gnu/packages/golang.scm (go-lukechampine-com-blake3): Move from here ...
* gnu/packages/golang-crypto.scm: ... to here.

* gnu/packages/uucp.scm: Add (gnu packages golang-crypto) to used modules.

Change-Id: I56dba8380317a0f58f92140889611ac3075af2cf
---
 gnu/packages/golang-crypto.scm | 26 ++++++++++++++++++++++++++
 gnu/packages/golang.scm        | 27 ---------------------------
 gnu/packages/uucp.scm          |  1 +
 3 files changed, 27 insertions(+), 27 deletions(-)

diff --git a/gnu/packages/golang-crypto.scm b/gnu/packages/golang-crypto.scm
index 91def02194..bfeea103ca 100644
--- a/gnu/packages/golang-crypto.scm
+++ b/gnu/packages/golang-crypto.scm
@@ -2,6 +2,7 @@
 ;;; Copyright © 2018 Pierre Neidhardt <mail@ambrevar.xyz>
 ;;; Copyright © 2019 Vagrant Cascadian <vagrant@debian.org>
 ;;; Copyright © 2019, 2020 Leo Famulari <leo@famulari.name>
+;;; Copyright © 2021 Arun Isaac <arunisaac@systemreboot.net>
 ;;; Copyright © 2021 Raghav Gururajan <rg@raghavgururajan.name>
 ;;; Copyright © 2022 (unmatched-parenthesis <paren@disroot.org>
 ;;; Copyright © 2022, 2023 Nicolas Graves <ngraves@ngraves.fr>
@@ -612,6 +613,31 @@ (define-public go-gitlab-com-yawning-edwards25519-extra
 possible.")
       (license license:bsd-3))))
 
+(define-public go-lukechampine-com-blake3
+  (package
+    (name "go-lukechampine-com-blake3")
+    (version "1.1.5")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/lukechampine/blake3")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1yxdwp8dpnnq2wbwsxlkbq570i99sc6781y39czjxi9jh9z5nw55"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "lukechampine.com/blake3"))
+    (propagated-inputs
+     (list go-github-com-klauspost-cpuid))
+    (home-page "https://pkg.go.dev/lukechampine.com/blake3")
+    (synopsis "Implementation of the BLAKE3 cryptographic hash function")
+    (description "@code{blake3} implements the BLAKE3 cryptographic hash
+function.  In addition to the pure-Go implementation, this package also
+contains AVX-512 and AVX2 routines (generated by avo) that greatly increase
+performance for large inputs and outputs.")
+    (license license:expat)))
 
 ;;;
 ;;; Avoid adding new packages to the end of this file. To reduce the chances
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 562d1adcfa..f6bec6c1c8 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -10429,33 +10429,6 @@ (define-public go-github-com-dustin-go-humanize
 back.")
     (license license:expat)))
 
-(define-public go-lukechampine-com-blake3
-  (package
-    (name "go-lukechampine-com-blake3")
-    (version "1.1.5")
-    (source
-     (origin
-       (method git-fetch)
-       (uri (git-reference
-             (url "https://github.com/lukechampine/blake3")
-             (commit (string-append "v" version))))
-       (file-name (git-file-name name version))
-       (sha256
-        (base32
-         "1yxdwp8dpnnq2wbwsxlkbq570i99sc6781y39czjxi9jh9z5nw55"))))
-    (build-system go-build-system)
-    (arguments
-     '(#:import-path "lukechampine.com/blake3"))
-    (propagated-inputs
-     (list go-github-com-klauspost-cpuid))
-    (home-page "https://pkg.go.dev/lukechampine.com/blake3")
-    (synopsis "Implementation of the BLAKE3 cryptographic hash function")
-    (description "@code{blake3} implements the BLAKE3 cryptographic hash
-function.  In addition to the pure-Go implementation, this package also
-contains AVX-512 and AVX2 routines (generated by avo) that greatly increase
-performance for large inputs and outputs.")
-    (license license:expat)))
-
 (define-public go-golang-org-x-term
   (package
     (name "go-golang-org-x-term")
diff --git a/gnu/packages/uucp.scm b/gnu/packages/uucp.scm
index 1626c90c36..37dc27625a 100644
--- a/gnu/packages/uucp.scm
+++ b/gnu/packages/uucp.scm
@@ -19,6 +19,7 @@
 
 (define-module (gnu packages uucp)
   #:use-module (gnu packages golang)
+  #:use-module (gnu packages golang-crypto)
   #:use-module (gnu packages golang-web)
   #:use-module (gnu packages texinfo)
   #:use-module (guix licenses)
-- 
2.41.0

